File: examples/javascript/qns_resolve.js
#!/usr/bin/env node
/**
 * Example 3 — QNS name resolution (QCore.js)
 *
 * Resolve a Quantova name under the capital Q top level domain to a Q1 address
 * through the on chain QNS registry container.
 *
 * Run:
 *   export QUANTOVA_GATEWAY=http://127.0.0.1:8645
 *   node qns_resolve.js
 *
 * In a full app with QCore.js installed:
 *   const { QCore } = require('@qunatovainc/qcore');
 *   const q = new QCore(process.env.QUANTOVA_GATEWAY);
 *   const qns = q.qns(QNS_REGISTRY);
 *   const addr = await qns.resolve('alice.Q');     // -> a Q1 address or null
 *   const owner = await qns.owner('alice.Q');
 */
'use strict';
const { gw, banner } = require('./_common');

const QNS_REGISTRY = 'Q1SNR7WZ2K9V5C3XM8H6U0AGNE2STV4YP93QDF';   // example QNS registry container
const NAME = 'alice.Q';

async function main() {
  banner('Quantova example: QNS name resolution (QCore.js)');

  const head = await gw('head');
  console.log(`Head height: ${head.height}`);
  console.log(`QNS registry: ${QNS_REGISTRY}`);

  // QCore.js does the name lookup inside qns.resolve(). Here we read the registry
  // storage directly to show the resolved record.
  const slot = await gw('get_storage', { address: QNS_REGISTRY, key: 'resolve:' + NAME });
  console.log(`\nresolve('${NAME}')  -> ${slot.value || null}`);

  console.log("\nQNS maps human names under the capital Q top level domain to Quantova");
  console.log("accounts through the on chain registry container. QCore.js exposes this");
  console.log("as q.qns(registry).resolve('alice.Q').");
}

main().catch((e) => { console.error('error:', e.message); process.exit(1); });

File: examples/python/qns_resolve.py
#!/usr/bin/env python3
"""Example 3 — QNS name resolution (QCore.py)

Resolve a Quantova name under the capital Q top level domain to a Q1 address
through the on chain QNS registry container, look up its owner, and reverse resolve
an address back to a name.

Run:
    export QUANTOVA_GATEWAY=http://127.0.0.1:8645
    python qns_resolve.py
"""

from _common import banner, connect

QNS_REGISTRY = "Q1SNR7WZ2K9V5C3XM8H6U0AGNE2STV4YP93QDF"  # example QNS registry container
NAME = "alice.Q"


def main():
    banner("Quantova example: QNS name resolution (QCore.py)")
    q = connect()
    print(f"Head height: {q.head()['height']}")

    qns = q.qns(QNS_REGISTRY)
    print(f"QNS registry: {QNS_REGISTRY}")

    addr = qns.resolve(NAME)
    print(f"\nresolve('{NAME}')  -> {addr}")

    try:
        owner = qns.owner(NAME)
        print(f"owner('{NAME}')    -> {owner}")
    except Exception as e:
        print(f"owner('{NAME}')    -> (not available: {e})")

    if addr:
        try:
            name = qns.reverse(addr)
            print(f"reverse({addr[:12]}...) -> {name}")
        except Exception as e:
            print(f"reverse(...)       -> (not available: {e})")

    print("\nQNS maps human names under the capital Q top level domain to Quantova")
    print("accounts through the on chain registry container.")


if __name__ == "__main__":
    main()
